### PR TITLE
[AKS] `az aks nodepool`: Allow specifying `--scale-down-mode` in nodepool create and update

### PR DESCRIPTION
**Pull request by marwanad**
**Description**

Brings in scale down mode for nodepools into CLI core.

Preview PR here: https://github.com/Azure/azure-cli-extensions/pull/3806/files

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change.
[AKS] `az aks nodepool`: Allow specifying `--scale-down-mode` in nodepool create and update 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Review by FumingZhang — APPROVED** [review]
LGTM :shipit:

**Comment by marwanad** [comment]
@zhoxing-ms can we get this merged before the CLI cut-off? Thanks!

File: src/azure-cli/azure/cli/command_modules/acs/_help.py [hunk 1]
```diff
@@ -905,6 +905,9 @@
   - name: --max-count
     type: int
     short-summary: Maximum nodes count used for autoscaler, when "--enable-cluster-autoscaler" specified. Please specify the value in the range of [1, 100]
+  - name: --scale-down-mode
+    type: string
+    short-summary: "Describes how VMs are added to or removed from nodepools."
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acs/_help.py:910]: Please use the first person voice
```suggestion
    short-summary: "Describe how VMs are added to or removed from nodepools."
```

File: src/azure-cli/azure/cli/command_modules/acs/_help.py [hunk 2]
```diff
@@ -1012,6 +1015,9 @@
...
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acs/_help.py:1020]:
```suggestion
    short-summary: "Describe how VMs are added to or removed from nodepools."
```

File: src/azure-cli/azure/cli/command_modules/acs/_help.py (hunk 1 above)
Comment (marwanad) [reply src/azure-cli/azure/cli/command_modules/acs/_help.py:910]: it should be "Describes". Short for "The parameter describes how VMs are added to or removed from nodepools"

**Comment by qpetraroia** [comment]
@zhoxing-ms Is it possible to get this merged before the deadline?

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@qpetraroia This PR has been merged

**Pull request merged**